### ISSUE TITLE
False positive: 'color-contrast' rule fails text nested inside multiple elements floating left

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#2222]
```color-contrast``` fails to pick up the background color of the text inside nested elements floating left and calculates the contrast ratio against the background of the next element flowing on the right side.

Expectation: ```color-contrast``` rule to pass the following example

Actual: ```color-contrast``` is being calculated against the wrong background color

Motivation: The failed text has sufficient contrast against its background

Demo link: https://codepen.io/iamrafan/pen/mdexPgw?editors=1100

<pre><code>axe-core version: 3.5.2-canary.37be85f
axe-Coconut version: 4.4.1</code></pre>

**Comment by straker** [comment]
Verified. Thanks for finding the issue.

**Comment by straker** [comment]
Putting a note here for later: The issue seems to be caused when we `visuallySort` the element stack. Our algorithm looks at the DOM position of the nodes and inverse sorts them based on DOM order. The issue is when the nested `div#mainContent > div` is compared against the `div#mainContent + div` element, the nested div returns as coming before in the DOM order so is sorted after the other div.

An idea to fix is to not compare different trees to one another since the parents of these trees determine order more than individual nodes.

**Comment by smth** [comment]
In case you want some more examples, my [homepage ](https://smth.uk/) seems to have 37 false positives :joy:

**Comment by padmavemulapati** [comment]
Verified, color-contrast calculation for children elements (floated - which are div2 div3 in the following example: 
`<div id="1" style="float: left; background: #000000; color: #fff;"> 
		<div id="2"><span id="target">whole picture</span></div> 
		</div> 
		<div id="3"> 
		<div id="4" style="background: #f2f2f2;">English</div> 
		</div>;`
working as expected.
![image](https://user-images.githubusercontent.com/47409579/82215101-747e3a00-9934-11ea-99ad-0904798b0692.png)

**Comment by jeankaplansky** [comment]
@straker - is there anything here we need to address in the rule help page for this rule?

**Comment by straker** [comment]
@jeankaplansky Nope, we can close.
